dashboard/lib/controllers/api/v1/pd/workshop_score_summarizer.rb

```diff
@@ -0,0 +1,74 @@
+module WorkshopScoreSummarizer
+  FACILITATOR_EFFECTIVENESS_QUESTIONS = [
+    :how_much_learned_s,
+    :how_motivating_s,
+    :how_clearly_presented_s,
+    :how_interesting_s,
+    :how_often_given_feedback_s,
+    :how_comfortable_asking_questions_s,
+    :how_often_taught_new_things_s
+  ]
+
+  TEACHER_ENGAGEMENT_QUESTIONS = [
+    :how_much_participated_s,
+    :how_often_talk_about_ideas_outside_s,
+    :how_often_lost_track_of_time_s,
+    :how_excited_before_s,
+    :overall_how_interested_s
+  ]
+
+  OVERALL_SUCCESS_QUESTIONS = [
+    :more_prepared_than_before_s,
+    :know_where_to_go_for_help_s,
+    :suitable_for_my_experience_s,
+    :would_recommend_s,
+    :part_of_community_s
+  ]
+
+  # There's some questions where a high number index is a bad thing (how often did you lose track of time?)
+  INVERTED_RESPONSE_QUESTIONS = [
+    :how_often_lost_track_of_time_s
+  ]
+
+  def get_score_for_workshops(workshops)
+    report_column = Hash.new(0)
+    response_count = 0
+
+    workshops.flat_map(&:enrollments).each do |enrollment|
+      response_object = PEGASUS_DB[:forms].where(source_id: enrollment.id, kind: 'PdWorkshopSurvey').first
```

Comment:
Rather than hitting the DB once per survey (how many enrollments are being iterated over), does it make sense to retrieve all the surveys in a single DB query?